File: July2022/Java/SearchA2DMatrixII.java

// @saorav21994
// TC : O(m+n)
// SC : O(1)

class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        
        int m = matrix.length;
        int n = matrix[0].length;
        
        int i = 0, j = n-1;
        
        while (i < m && j >= 0) {
            // System.out.println(matrix[i][j]);
            int val = matrix[i][j];
            if (val == target)
                return true;
            if (target < val)
                j -= 1;
            else
                i += 1;
        }
        
        return false;
        
    }
}

File: July2022/Python/searchA2DMatrixII.py
@ Jiganesh
# https://leetcode.com/problems/search-a-2d-matrix-ii/

from typing import List



class Solution:

    # TC : O(m*n)
    # SC : O(1)
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        row , col = 0, len(matrix[0])-1
        while row<len(matrix) and  col>=0:
        
            pointing = matrix[row][col]
            if pointing == target : return True
            if pointing > target: col-=1
            else : row+=1
                
        return False